test/jdk/java/nio/channels/Channels/ReadXBytes.java

```diff
@@ -138,6 +148,22 @@ public void dataTest(long length, FileCreator c, DataTest f)
         }
     }
 
+    // Runs test against an InputStream where the FileChannel
+    // relationship is obscured.
+    public void dataTestWrappedInputStream(long length, FileCreator c,
+        DataTest f) throws IOException {
+        Path file = c.create(length);
+        try (FileInputStream fis = new FileInputStream(file.toFile());
+             FileInputStream fis2 = new FileInputStream(file.toFile());
+             ReadableByteChannel ch = Channels.newChannel(
+                 new FilterInputStream(fis2) {});
+             InputStream cis = Channels.newInputStream(ch)) {
```

Comment:
With four resources in the `try`, this `try-with-resources` is a little hard to read. Perhaps use two nested `t-w-r`s instead?